src/watcher.js

```diff
@@ -115,41 +115,176 @@ async function getLastBlockToProcess() {
+        event: config.event,
+        fromBlock,
+        toBlock,
+        filter: config.eventFilter
+      })
+      logger.info(
+        `Found ${events.length} ${config.event} events for contract address
+          ${eventContract.options.address}`
+      )
 
-    if (events.length) {
-      const job = await processEvents(events)
-      logger.info('Transactions to send:', job.length)
+      await processBridgeMappingsAdded(events)
 
-      if (job.length) {
-        await sendToQueue(job)
-      }
+      logger.debug(
+        { lastProcessedBlock: lastBlockToProcess.toString() },
+        'Updating last processed block'
+      )
+      await updateLastProcessedBlock(lastBlockRedisKey, lastBlockToProcess)
+    }
+  } else {
+    const mapper = async bridgeObj => {
+      const homeStartBlock = toBN(bridgeObj.homeStartBlock || 0)
+      const foreignStartBlock = toBN(bridgeObj.foreignStartBlock || 0)
+      const bridgeContractAddress =
+        config.id === 'erc-erc-multiple-affirmation-request'
+          ? bridgeObj.foreignBridge
+          : bridgeObj.homeBridge
+      const eventContractAddress =
+        config.id === 'erc-erc-multiple-affirmation-request'
+          ? bridgeObj.foreignToken
+          : bridgeContractAddress
+      const eventFilter =
+        config.id === 'erc-erc-multiple-affirmation-request' ? { to: bridgeObj.foreignBridge } : {}
+      const bridgePair = `${bridgeObj.homeBridge}:${bridgeObj.foreignBridge}`
+      const lastBlockRedisKey = `${lastBlockRedisKeyDefault}:${bridgePair}`
+      const lastProcessedBlock = await getLastProcessedBlock(
+        lastBlockRedisKey,
+        config.id === 'erc-erc-multiple-affirmation-request'
+          ? BN.max(foreignStartBlock.sub(ONE), ZERO)
+          : BN.max(homeStartBlock.sub(ONE), ZERO)
+      )
+      const lastBlockToProcess = await processOne(
+        sendToQueue,
+        bridgeContractAddress,
+        eventContractAddress,
+        bridgeObj.homeBridge,
+        bridgeObj.foreignBridge,
+        eventFilter,
+        lastProcessedBlock
+      )
+      return { lastBlockToProcess, lastBlockRedisKey }
     }
 
-    logger.debug(
-      { lastProcessedBlock: lastBlockToProcess.toString() },
-      'Updating last processed block'
-    )
-    await updateLastProcessedBlock(lastBlockToProcess)
+    const deployedBridges = await getDeployedBridges()
+    logger.debug(`Found ${deployedBridges.length} deployed bridges`)
+    const results = await processConcurrently(deployedBridges, mapper, config.concurrency)
+    results.forEach(async res => {
+      logger.debug(
+        { lastBlockToProcess: res.lastBlockToProcess.toString() },
+        `Updating last processed block for key ${res.lastBlockRedisKey}`
+      )
+      await updateLastProcessedBlock(res.lastBlockRedisKey, res.lastBlockToProcess)
+    })
+  }
+}
+
+async function main({ sendToQueue }) {
+  try {
+    if (config.id.startsWith('erc-erc-multiple')) {
+      processErcToErcMultiple(sendToQueue)
```

Comment:
We should wait for `processErcToErcMultiple` to finish processing
```suggestion
      await processErcToErcMultiple(sendToQueue)
```